### PR TITLE
Add Zoned Storage libraries + helper executables

### PR DESCRIPTION
**Pull request by fogti**
###### Description of changes

Overview page: https://zonedstorage.io/docs/tools

https://github.com/westerndigitalcorporation/libzbc
https://github.com/westerndigitalcorporation/libzbd
https://github.com/linux-nvme/libnvme

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/libnvme/default.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  postPatch = ''
+    chmod +x doc/kernel-doc-check
+    patchShebangs .
+  '';
+
+  doCheck = true;
+
+  outputs = ["out" "man"];
+
+  meta = {
```
Comment (azahi) [inline pkgs/os-specific/linux/libnvme/default.nix:55]: Please do this for `meta`.
```suggestion
  meta = with lib; {
```

**Review by azahi — CHANGES_REQUESTED** [review]
Please don't use `alejandra` to format code in Nixpkgs. If you need a formatter, consider using `nixpkgs-fmt`. IIRC there's no "official" style reference, so the safest bet for new contributions would be to mimic what's already in Nixpkgs. So, please reformat your code so that it looks in line with other derivations.

File: pkgs/os-specific/linux/libnvme/default.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  libuuid,
+  meson,
+  ninja,
+  openssl,
+  perl,
+  pkg-config,
+  python3,
+  stdenv,
+  systemd,
+}:
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/libnvme/default.nix:14]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
, json_c
, libuuid
, meson
, ninja
, openssl
, perl
, pkg-config
, python3
, systemd
}:

```

File: pkgs/os-specific/linux/libnvme/default.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  mesonFlags = [
+    "-Ddocs=man"
+    "-Ddocs-build=true"
+  ];
+
+  postPatch = ''
+    chmod +x doc/kernel-doc-check
+    patchShebangs .
+  '';
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/libnvme/default.nix:49]: Canonically we write postPatch after src. Also be more specific about the files that need patching.

File: pkgs/os-specific/linux/libnvme/default.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+  doCheck = true;
+
+  outputs = ["out" "man"];
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/libnvme/default.nix:53]: outputs is normally put between version and src.

File: pkgs/os-specific/linux/libnvme/default.nix [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    "-Ddocs=man"
+    "-Ddocs-build=true"
+  ];
+
+  postPatch = ''
+    chmod +x doc/kernel-doc-check
+    patchShebangs .
+  '';
+
+  doCheck = true;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/libnvme/default.nix:51]: Isn't this the default?

File: pkgs/os-specific/linux/libnvme/default.nix [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+  outputs = ["out" "man"];
+
+  meta = {
+    description = "C Library for NVM Express on Linux";
+    homepage = "https://github.com/linux-nvme/libnvme";
+    maintainers = with lib.maintainers; [zseri];
+    license = with lib.licenses; [lgpl21Plus];
+    platforms = lib.platforms.linux;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/libnvme/default.nix:60]:
```suggestion
    maintainers = with maintainers; [ zseri ];
    license = with licenses; [ lgpl21Plus ];
    platforms = platforms.linux;
```

File: pkgs/os-specific/linux/libzbc/default.nix [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  autoconf-archive,
+  autoreconfHook,
+  fetchFromGitHub,
+  gtk3,
+  lib,
+  libtool,
+  pkg-config,
+  stdenv,
+  guiSupport ? false,
+}:
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/libzbc/default.nix:11]:
```suggestion
{ lib
, stdenv
, autoconf-archive
, autoreconfHook
, fetchFromGitHub
, gtk3
, libtool
, pkg-config
, guiSupport ? false
}:

```

File: pkgs/os-specific/linux/libzbc/default.nix [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    sha256 = "qI09dkMCwMym3j1ELrFDNbNB5hW/CzwmFmZhUNDXsfI=";
+  };
+
+  nativeBuildInputs = [
+    autoconf-archive # this can be removed with the next release
+    autoreconfHook
+    libtool
+  ];
+
+  buildInputs = lib.optionals guiSupport [gtk3 pkg-config];
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/libzbc/default.nix:29]:
```suggestion
  buildInputs = lib.optionals guiSupport [ gtk3 pkg-config ];
```
pkg-config goes to nativeBuildInputs.

File: pkgs/os-specific/linux/libzbd/default.nix [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+  autoreconfHook,
+  fetchFromGitHub,
+  gtk3,
+  lib,
+  libtool,
+  pkg-config,
+  stdenv,
+  guiSupport ? false,
+}:
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/libzbd/default.nix:12]:
```suggestion
{ lib
, stdenv
, autoconf-archive
, autoreconfHook
, fetchFromGitHub
, gtk3
, libtool
, pkg-config
, guiSupport ? false
}:

stdenv.mkDerivation rec {
```

File: pkgs/os-specific/linux/libzbd/default.nix [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    sha256 = "GoCHwuR4ylyaN/FskIqKyAPe2A2O3iFVcI3UxPlqvtk=";
+  };
+
+  nativeBuildInputs = [
+    autoconf-archive # this can be removed with the next release
+    autoreconfHook
+    libtool
+  ];
+
+  buildInputs = lib.optionals guiSupport [gtk3 pkg-config];
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/libzbd/default.nix:29]:
```suggestion
  buildInputs = lib.optionals guiSupport [ gtk3 ];
```
pkg-config goes to nativeBuildInputs.

File: pkgs/os-specific/linux/libzbc/default.nix [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+
+  configureFlags = lib.optional guiSupport "--enable-gui";
+
+  meta = {
+    description = "ZBC device manipulation library";
+    homepage = "https://github.com/westerndigitalcorporation/libzbc";
+    maintainers = with lib.maintainers; [zseri];
+    license = with lib.licenses; [bsd2 lgpl3Plus];
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/libzbc/default.nix:37]:
```suggestion
    maintainers = with lib.maintainers; [ zseri ];
    license = with lib.licenses; [ bsd2 lgpl3Plus ];
```

File: pkgs/os-specific/linux/libzbd/default.nix [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+
+  configureFlags = lib.optional guiSupport "--enable-gui";
+
+  meta = {
+    description = "Zoned block device manipulation library and tools";
+    homepage = "https://github.com/westerndigitalcorporation/libzbd";
+    maintainers = with lib.maintainers; [zseri];
+    license = with lib.licenses; [lgpl3Plus gpl3Plus];
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/libzbd/default.nix:37]:
```suggestion
    maintainers = with lib.maintainers; [ zseri ];
    license = with lib.licenses; [ lgpl3Plus gpl3Plus ];
```

**Review by SuperSandro2000 — COMMENTED** [review]
FYI using alejandra generates code that is wrongly formatted for nixpkgs.

File: pkgs/os-specific/linux/libnvme/default.nix (hunk 5 above)
Comment (zseri) [reply pkgs/os-specific/linux/libnvme/default.nix:51]: https://github.com/NixOS/nixpkgs/blob/5518e6aba11f50f0953fa21c7ad6a283c4741b95/pkgs/stdenv/generic/make-derivation.nix#L121

**Comment by zseri** [comment]
@azahi done

**Comment by azahi** [comment]
Result of `nixpkgs-review pr 189672` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>libnvme</li>
    <li>libzbc</li>
    <li>libzbd</li>
  </ul>
</details>

**Review by azahi — APPROVED** [review]

**Pull request merged**